src/neuroglancer/ui/annotations.ts

```diff
@@ -1316,7 +1317,12 @@ export function UserLayerWithAnnotationsMixin<TBase extends {new (...args: any[]
           !this.annotationStates.states.includes(annotationLayer)) {
         return;
       }
+      state.rank = mouseState.coordinateSpace.rank;
```

Comment:
This is the rank of the global coordinate space, which is not necessarily the same as the rank of the annotation source.  I think we don't actually need to store the rank here, instead you can obtain the rank from `annotationLayer.source.rank` inside of `displayAnnotationState`.

Reply:
Updated.